Connecting_MonteCarlo_to_ModernAI/N2_2D_Ising_Model.ipynb

```diff
@@ -0,0 +1,1233 @@
+      ],
+      "metadata": {
+        "id": "EVWnLZH9smZO"
+      }
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "#### Execution Optimization Toolkit\n",
+        "Here are some common tools useful for analyzing how your code runs. These tools focus on optimizing *speed* and *memory*; there are other tools for optimizing other aspects of programming like [documentation](https://docs.python.org/3/library/pydoc.html) and [code consistency](https://www.techbeamers.com/pylint-tool/).\n",
+        "\n",
+        "|What you are optimizing|What you use to optimize it| How it works\n",
+        "| --- | --- | --- |\n",
+        "|Speed|[timeit](https://docs.python.org/3/library/timeit.html)|Measures how long a code snippet takes to execute|\n",
+        "|Speed|[tqdm](https://tqdm.github.io/)|Measures how long a loop takes to execute *with a loading bar!!!*| \n",
+        "|Memory|[memory-profiler](https://pypi.org/project/memory-profiler/#description)|See which variables and programs use what memory resource |\n",
+        "\n",
+        "Note that these are diagnostic tools that provide a way to measure what the computer does with your code.  It is still up to you to find solutions to slow parts or out-of-memory (OOM) errors and implement them."
+      ],
+      "metadata": {
+        "id": "Z259mJbqjIQn"
+      }
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "# Exercises\n",
+        "\n",
+        "Complete the exercises below."
+      ],
+      "metadata": {
+        "id": "fx5V1t6PKCNE"
+      }
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "## Programming Exercise 1: Optimize the Speed of the Metropolis-Hastings Algorithm\n"
+      ],
+      "metadata": {
+        "id": "AiAPXGCSZIja"
+      }
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "### 1a) Complete the functions `intialize_lattice` and `MCMC_step` \n",
+        "\n",
+        "... using the information provided above and the commented hints in the function templates provided below.  Feel free to copy code from above as you fill in the blanks, but be careful!  Some variable names have been changed..."
+      ],
+      "metadata": {
+        "id": "a3tf6uQLlJXj"
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "# Complete the initialize_lattice function below\n",
+        "\n",
+        "def initialize_lattice(sqrt_n: int):\n",
+        "    \"\"\"    \n",
+        "    Function to initialize lattice.  Adds a border of zeros\n",
+        "    to represent non-interacting atoms and make the neighbor\n",
+        "    calculation easier\n",
+        "\n",
+        "    sqrt_n: The square root of the number of atoms in the lattice\n",
```

Comment:
I was a little confused by `sqrt_n` in the parameters and `L` in all the comments. Took me a second to realize they were the same. I'm not sure the best way to clarify..

Reply:
Thank you for your comment.  I have added wording explaining what `sqrt_n` is to the markup, and removed references to variable `L`.